core/src/main/java/org/apache/iceberg/util/PropertyUtil.java

```diff
@@ -120,6 +120,27 @@ public static Map<String, String> propertiesWithPrefix(
         .collect(Collectors.toMap(e -> e.getKey().replaceFirst(prefix, ""), Map.Entry::getValue));
   }
 
+  /**
+   * Returns subset of provided map with keys matching the provided prefix. Matching is
+   * case-sensitive.
+   *
+   * @param properties input map
+   * @param prefix prefix to choose keys from input map
+   * @return subset of input map with keys starting with provided prefix
+   */
+  public static Map<String, String> propertiesWithPrefixNoTrim(
```

Comment:
"Trim" typically means removing whitespace. Is there a better way to name this method? Maybe the `startsWith` could be passed as a `Predicate<String>`  and this could be `filterProperties`? Or maybe `filterPropertiesByPrefix`?